`Argument "headers" to "FieldStorage" has incompatible type `

**Bug Report**

Hi,
consider the following examples:

```python
import http.server
from cgi import FieldStorage
from http import HTTPStatus

class Handler(http.server.SimpleHTTPRequestHandler):
    def do_POST(self) -> None:
        if self.path == "/upload":
            result = self.receive_upload()
    def receive_upload(self) -> tuple[HTTPStatus, str]:
        result = (HTTPStatus.INTERNAL_SERVER_ERROR, "Server error")
        form = FieldStorage(fp=self.rfile, headers=self.headers, environ={"REQUEST_METHOD": "POST"})
        return result
```

I get:

```console
main.py:11: error: Argument "headers" to "FieldStorage" has incompatible type "Message"; expected "Optional[Mapping[str, str]]"
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

Run on <https://mypy-play.net/?mypy=latest&python=3.10>

Am I missing something?

Thanks

--- a/stdlib/cgi.pyi
+++ b/stdlib/cgi.pyi
@@ -1,9 +1,8 @@
-from email.message import Message
-
 import sys
 from _typeshed import Self, SupportsGetItem, SupportsItemAccess
 from builtins import list as _list, type as _type
 from collections.abc import Iterable, Iterator, Mapping
+from email.message import Message
 from types import TracebackType
 from typing import IO, Any, Protocol
 

--- a/stdlib/cgi.pyi
+++ b/stdlib/cgi.pyi
@@ -1,3 +1,5 @@
+from email.message import Message
+
 import sys
 from _typeshed import Self, SupportsGetItem, SupportsItemAccess
 from builtins import list as _list, type as _type
@@ -72,7 +74,7 @@ class FieldStorage:
     keep_blank_values: int
     strict_parsing: int
     qs_on_post: str | None
-    headers: Mapping[str, str]
+    headers: Mapping[str, str] | Message
     fp: IO[bytes]
     encoding: str
     errors: str
@@ -93,7 +95,7 @@ class FieldStorage:
     def __init__(
         self,
         fp: IO[Any] | None = ...,
-        headers: Mapping[str, str] | None = ...,
+        headers: Mapping[str, str] | Message | None = ...,
         outerboundary: bytes = ...,
         environ: SupportsGetItem[str, str] = ...,
         keep_blank_values: int = ...,

